ckanext/sso/plugin.py

```diff
@@ -45,44 +45,53 @@ def after_saml2_login(self, resp, saml_attributes):
         user_id = str(toolkit.g.userobj.id)
         user = model.User.by_name(username)
 
-        if is_ckan_admin:
+        if is_ckan_admin: # add sysadmin privileges
             user.sysadmin = True
             model.Session.add(user)
             model.Session.commit()
             return resp
+        else: # remove sysadmin privileges
+            user.sysadmin = False
+            model.Session.add(user)
+            model.Session.commit()
 
-        # remove sysadmin privileges
-        user.sysadmin = False
-        model.Session.add(user)
-        model.Session.commit()
-
-        # Check if their organisation exist. If it does, add them to it at assigned level
-        try:
-            expected_org_title = saml_attributes["member"][0] # use the first organization in the list only
-            expected_org_name = expected_org_title.replace(' ', '-').lower()
-        except (KeyError, IndexError):
-            log.error("unexpected user with no group membership: %s", username)
-            return resp
+            # Remove from all orgs, Simpler logic than the other option.
```

Comment:
I agree, but detecting when the user is already a member can reduce the number of db trips required thus making it faster. Might be relevant since this is done every time a user logs in. It's up to you whether it's worth it or not.